stream_alert/athena_partition_refresh/main.py

```diff
@@ -0,0 +1,206 @@
+        ResultConfiguration={'OutputLocation': '{}/{}'.format(
+            kwargs['results_bucket'],
+            kwargs['results_path']
+        )}
+    )
+    query_execution_result = check_query_status(query_execution_resp['QueryExecutionId'])
+    if query_execution_result != 'SUCCEEDED':
+        LOGGER.error(
+            'The query %s returned %s, exiting!',
+            kwargs['query'],
+            query_execution_result)
+        return False
+
+    query_results_resp = ATHENA_CLIENT.get_query_results(
+        QueryExecutionId=query_execution_resp['QueryExecutionId'],
+    )
+
+    # The idea here is to leave the processing logic to the calling functions.
+    # No data being returned isn't always an indication that something is wrong.
+    if not query_results_resp['ResultSet']['Rows']:
+        LOGGER.debug('The query %s returned no data', kwargs['query'])
+
+    return query_results_resp
+
+
+def check_database_exists(results_bucket, results_path):
+    """Verify the StreamAlert Athena database exists."""
+    query_resp = run_athena_query(
+        query='SHOW DATABASES LIKE \'streamalert\';',
+        results_bucket=results_bucket,
+        results_path=results_path
+    )
+    if isinstance(query_resp, dict) and not query_resp['ResultSet']['Rows']:
+        LOGGER.info('The \'streamalert\' database does not exist, please create it.')
+        return False
+
+    return True
+
+
+def check_table_exists(results_bucket, results_path, table_name):
+    """Verify a given StreamAlert Athena table exists."""
+    query_resp = run_athena_query(
+        query='SHOW TABLES LIKE \'{}\';'.format(table_name),
+        database='streamalert',
+        results_bucket=results_bucket,
+        results_path=results_path
+    )
+    if isinstance(query_resp, dict) and not query_resp['ResultSet']['Rows']:
+        LOGGER.info('The streamalert table \'%s\' does not exist, please create it.', table_name)
+        return False
+
+    return True
+
+
+def normal_partition_refresh(config, athena_results_bucket, athena_results_path):
+    normal_partition_config = config['lambda']['athena_partition_refresh_config']['partitioning']['normal']
+    for _, athena_table in normal_partition_config.iteritems():
+        resp = run_athena_query(
+            query='MSCK REPAIR TABLE {};'.format(athena_table),
+            database='streamalert',
+            results_bucket=athena_results_bucket,
+            results_path=athena_results_path
+        )
+        if resp:
+            LOGGER.info('Query results:')
+            for row in resp['ResultSet']['Rows']:
+                LOGGER.info(row)
+
+
+def firehose_partition_refresh(_):
+    LOGGER.error('Firehose partition refresh is not yet supported, exiting!')
+    raise NotImplementedError
+
+
+def handler(event, context):
+    """Athena Partition Refresher Handler Function"""
+    config = _load_config()
+    if not config:
+        LOGGER.error('No config found, exiting!')
```

Comment:
It's also not really true; `_load_config` returns `None` if there was an error in parsing. If the config file doesn't exist, it throws an uncaught IOError.

Since an error reading the config file is so damaging that the entire function has to halt its execution, this is a great example of an error which should bubble all the way up, breaking the function execution. In other words, instead of trying to catch `IOError` and/or a parsing error, just let config errors bubble up. The code is simpler and the error will be caught faster